src/modules/esl-carousel/plugin/dots/esl-carousel.nav.dots.ts

```diff
@@ -0,0 +1,191 @@
+    dot.setAttribute('role', 'tab');
+    dot.setAttribute('tabindex', tabIndex >= 0 ? '-1' : '0');
+    dot.setAttribute('esl-carousel-dot', `${index}`);
+    dot.setAttribute('aria-label', format(dotLabelFormat, {index: index + 1}));
+    return dot;
+  }
+  /** Default dot updater implementation (readonly)*/
+  public static defaultDotUpdater($dot: HTMLElement, active: boolean): void {
+    $dot.toggleAttribute('active', active);
+    $dot.setAttribute('aria-disabled', String(!active));
+    $dot.setAttribute('aria-current', String(active));
+  }
+
+  /** Default dots builder function {@link ESLCarouselNavDotsBuilder} */
+  public static dotsBuilder: ESLCarouselNavDotsBuilder = ESLCarouselNavDots.defaultDotBuilder;
+  /** Default dots updater function {@link ESLCarouselNavDotsUpdater} */
+  public static dotsUpdater: ESLCarouselNavDotsUpdater = ESLCarouselNavDots.defaultDotUpdater;
+
+  /** {@link ESLTraversingQuery} string to find {@link ESLCarousel} instance */
+  @attr({
+    name: 'target',
+    defaultValue: '::parent([esl-carousel-container])::find(esl-carousel)'
+  })
+  public carousel: string;
+
+  /** Label format (supports `index` key) */
+  @attr({defaultValue: ($this: ESLCarouselNavDots) => `Go to slide ${$this.groupSize > 1  ? 'group ' : ''}{index}`})
+  public dotLabelFormat: string;
+
+  /** Dots builder function {@link ESLCarouselNavDotsBuilder} */
+  @prop(ESLCarouselNavDots.dotsBuilder) public dotsBuilder: ESLCarouselNavDotsBuilder;
+  /** Dots updater function {@link ESLCarouselNavDotsUpdater} */
+  @prop(ESLCarouselNavDots.dotsUpdater) public dotsUpdater: ESLCarouselNavDotsUpdater;
+
+  public get groupSize(): number {
+    return this.$carousel?.state.count || 0;
+  }
+
+  /** Dots number according carousel config */
+  public get count(): number {
+    if (!this.$carousel) return 0;
+    const {count, size} = this.$carousel.state;
+    return Math.ceil(size / count);
+  }
+
+  /** Active dot index according to carousel config */
+  public get activeIndex(): number {
+    if (!this.$carousel) return 0;
+    const {activeIndex, count, size} = this.$carousel.state;
+    return indexToGroup(activeIndex, count, size);
+  }
+  /** Previous dot index (cycled) */
+  public get prevIndex(): number {
+    return this.activeIndex > 0 ? this.activeIndex - 1 : this.count - 1;
+  }
+  /** Next dot index (cycled) */
+  public get nextIndex(): number {
+    return this.activeIndex < this.count - 1 ? this.activeIndex + 1 : 0;
+  }
+
+  /** Current dots collection */
+  public get $dots(): HTMLElement[] {
+    return [...this.querySelectorAll('[esl-carousel-dot]')] as HTMLElement[];
+  }
+
+  /** @returns ESLCarousel instance; based on {@link carousel} attribute */
+  @memoize()
+  public get $carousel(): ESLCarousel | null {
+    return ESLTraversingQuery.first(this.carousel, this) as ESLCarousel;
+  }
+
+  public override async connectedCallback(): Promise<void> {
+    this.$$attr('disabled', true);
```

Comment:
What's the purpose of this?

Reply:
Updated: 'disabled' marker is used to indicate dots passive state (when amount of dots is less or equal 1)

Reply:
Updated once more, I probably use native empty and do not render fictive state